interfaces/builtin/maliit_input_method.go

```diff
@@ -0,0 +1,170 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const maliitInputMethodPermanentSlotAppArmor = `
+# Description: Allow operating as a maliit server.
+# Communication with maliit happens in the following stages:
+#  * An application connects to the address service: org.maliit.Server.Address.
+#  * The server responds with a private unix socket of the form
+#    @/tmp/maliit-server/dbus-* on which the server is running a peer-to-peer
+#    dbus session.
+#  * All further communication happens over this channel
+#  * An application wishing to receive input then requests that it be made the
+#    active context.
+#  * At this point maliit retrieves the application's PID based on the dbus
+#    channel and verifies with Unity 8 that the application is currently
+#    focused.
```

Comment:
@elleo I think it is acceptable for the app to tell us their surface ID because Mir ultimately decides if the surface ID is focused when maliit calls isSurfaceFocused(). As of last time I looked at it, the surface ID was randomly generated and was not brute forceable in any reasonable amount of time so an untrusted client application couldn't give valid, in-focus surface IDs. The first two paragraphs in the link below are relevant:

  https://code.launchpad.net/~ken-vandine/content-hub/pasteboard/+merge/296352/comments/785060